### PR TITLE
Adding EnvironmentVersionTag resource

### PR DESCRIPTION
**Pull request by IaroslavTitov**
### Summary
- Adding EnvironmentVersionTag resource

### Testing
- Tested locally using Dotnet SDK

```
    var environ = new Pulumi.PulumiService.Environment(
        "Iaro's environment",
        new EnvironmentArgs {
            Organization = "service-provider-test-org",
            Name = "IaroEnv",
            Yaml = new StringAsset(yaml)
        }
    );

    var tag = new Pulumi.PulumiService.EnvironmentVersionTag(
        "Iaro's version tag",
        new EnvironmentVersionTagArgs {
            Organization = "service-provider-test-org",
            Environment = "IaroEnv",
            TagName = "SuperTag",
            Revision = environ.Revision
        }
    );
```

### CODE REVIEW

File: provider/pkg/provider/environment_version_tags.go [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+	}, nil
+}
+
+func (evt *PulumiServiceEnvironmentVersionTagResource) Check(req *pulumirpc.CheckRequest) (*pulumirpc.CheckResponse, error) {
+	return &pulumirpc.CheckResponse{Inputs: req.News, Failures: nil}, nil
+}
+
+func (evt *PulumiServiceEnvironmentVersionTagResource) Update(req *pulumirpc.UpdateRequest) (*pulumirpc.UpdateResponse, error) {
+	// all updates are destructive, so we just call Create.
+	return nil, fmt.Errorf("unexpected call to update, expected create to be called instead")
```
Comment (pgavlin) [inline provider/pkg/provider/environment_version_tags.go:118]: We definitely don't want tags to always be replaced. Every change to a tag besides its name should be an update.

**Review by pgavlin — CHANGES_REQUESTED** [review]

File: provider/pkg/provider/environment_version_tags.go (hunk 1 above)
Comment (IaroslavTitov) [reply provider/pkg/provider/environment_version_tags.go:118]: I followed the way stack tags are implemented [here](https://github.com/pulumi/pulumi-pulumiservice/blob/f13da7e94e5f83093da265d3a3c7c4d842375ba9/provider/pkg/provider/stack_tags.go#L133)

I'll rework it to have updates

**Review by pgavlin — APPROVED** [review]

File: provider/cmd/pulumi-resource-pulumiservice/schema.json [hunk 2]
```diff
@@ -1344,6 +1344,33 @@
         "environment",
         "permission"
       ]
+    },
+    "pulumiservice:index:EnvironmentVersionTag": {
```
Comment (komalali) [inline provider/cmd/pulumi-resource-pulumiservice/schema.json:1348]: Does the resource not have any outputs?

Comment (komalali) [reply provider/cmd/pulumi-resource-pulumiservice/schema.json:1348]: We should mirror all inputs as outputs, at least

Comment (IaroslavTitov) [reply provider/cmd/pulumi-resource-pulumiservice/schema.json:1348]: Good catch, somehow works without it lol
I'll add real quick

**Review by komalali — APPROVED** [review]

**Pull request merged**